src/crystal/system/win32/thread.cr

```diff
@@ -16,42 +17,91 @@ class Thread
   property previous : Thread?
 
   def self.unsafe_each
-    @@threads.unsafe_each { |thread| yield thread }
+    threads.unsafe_each { |thread| yield thread }
   end
 
+  # Starts a new system thread.
+  def initialize(&@func : ->)
+    @th = uninitialized LibC::HANDLE
+
+    @th = GC.beginthreadex(
+      security: Pointer(Void).null,
+      stack_size: LibC::UInt.zero,
+      start_address: ->(data : Void*) { data.as(Thread).start; LibC::UInt.zero },
+      arglist: self.as(Void*),
+      initflag: LibC::UInt.zero,
+      thrdaddr: Pointer(LibC::UInt).null)
+
+    if @th.null?
+      raise RuntimeError.from_errno("_beginthreadex")
```

Comment:
Hmm, raising on null both here and in `beginthreadex` seems a bit redundant.

Reply:
Indeed.